crates/core/src/util/lending_pool.rs

```diff
@@ -49,6 +49,35 @@ struct PoolVec<T> {
 #[derive(Debug)]
 pub struct PoolClosed;
 
+/// A scope guard for the reducer queue length metric,
+/// ensuring an increment is always be paired with one and only one decrement.
+struct QueueMetric {
+    db: Address,
+}
+
+impl Drop for QueueMetric {
+    fn drop(&mut self) {
+        WORKER_METRICS.instance_queue_length.with_label_values(&self.db).dec();
+        let queue_length = WORKER_METRICS.instance_queue_length.with_label_values(&self.db).get();
+        WORKER_METRICS
+            .instance_queue_length_histogram
+            .with_label_values(&self.db)
+            .observe(queue_length as f64);
```

Comment:
why do we need a separate histogram? shouldn't grafana accumulate the data for us over time anyway?

Reply:
if you didn't need to se the histogram you could just use the `inc_scope` utility function I made a while ago, which does this same sorta thing.

Reply:
> why do we need a separate histogram? shouldn't grafana accumulate the data for us over time anyway?

The `instance_queue_length` metric is a gauge which is only sampled once per sampling interval. So all it gives us is the queue length at the end of the sampling interval over time. But it doesn't tell us anything about the queue length at the beginning or middle of the sampling interval.

`instance_queue_length_histogram` on the other hand will build a histogram of all the different values the queue length takes throughout the sampling interval, and this will give us percentiles.

It is true that `instance_queue_length` really doesn't give us any new information over the histogram. We just need some variable to track the current length of the queue. I just chose a prometheus metric for that purpose mainly for ease of use.

Reply:
> you could just use the inc_scope utility function I made a while ago

I would have done that if it weren't for the `get`